cabal-install/src/Distribution/Client/ProjectBuilding/UnpackedPackage.hs

```diff
@@ -262,27 +270,30 @@ buildAndRegisterUnpackedPackage
         | hasValidHaddockTargets pkg = action
         | otherwise = return ()
 
+      mbWorkDir = useWorkingDir $ scriptOptions
```

Comment:
Why the `$`?

Reply:
As you've noted @geekosaur. This is one of those things we could be relying on HLint for, instead of checking for things like this manually at review, #9110;

```diff
git diff
diff --git a/.hlint.yaml b/.hlint.yaml
index e38cc7be7..7c9f4cbbd 100644
--- a/.hlint.yaml
+++ b/.hlint.yaml
@@ -13,7 +13,6 @@
 - ignore: {name: "Monoid law, left identity"} # 3 hints
 - ignore: {name: "Monoid law, right identity"} # 3 hints
 - ignore: {name: "Move guards forward"} # 4 hints
-- ignore: {name: "Redundant $"} # 125 hints
 - ignore: {name: "Redundant $!"} # 4 hints
 - ignore: {name: "Redundant <$>"} # 6 hints
 - ignore: {name: "Redundant =="} # 1 hint
```

```
$ hlint cabal-install/src/Distribution/Client/ProjectBuilding/UnpackedPackage.hs
cabal-install/src/Distribution/Client/ProjectBuilding/UnpackedPackage.hs:273:33: Suggestion: Redundant $
Found:
  useWorkingDir $ scriptOptions
Perhaps:
  useWorkingDir scriptOptions

1 hint
```